wordpress-mobile/WordPress-Android
# Site Creation: Site Info page improvements

Fixes #8736 & #8766. This PR contains a set of small improvements mostly to the site info screen as well as fixing our master branch's build which was broken due to merging `develop` into it. Here is a list of changes:

* We now request focus for the title text field as soon as the user enters site info page - technically, there is a `500ms` delay. It also adds a unit test for it.
* Fixes an issue in the segments page where an empty icon color was causing a crash. It also adds a `TODO` so we can check in with the designers on how best to handle this issue. a413760fa4d2498437f80864dff101f3b5c7a0a7
* Fixes the site info and vertical view model tests due to changes in FluxC
* Updates the site info page layout to keep the `skip/next` button visible at all times as described in #8736. It looks like the layout will be scrollable in most cases, so the shadow is visible at all times. This is consistent with the new domains page. We'll address any concerns about it during the final design review if necessary.
* Updates the `Optional Tagline` text to `Tagline` in site info page as described in #8766.

We'll be doing a full design review at the end of site creation project, so a design review is not required for this PR.

To test:
* Go to site info screen (check the gif below to see how to access it)
* Check that the `skip/next` button and its container is fixed to bottom and the content above it is scrollable
* Check that the `title` text field is selected and the keyboard shows up as you enter the screen.
* Feel free to run the unit tests that were changed added, but Travis already does it for us, so it's not necessary.

Here is a gif that's showcasing the changes and demonstrates how to access the site info screen: https://cloudup.com/cyPiVq-Zt2b

One reviewer should be enough for this PR, I only asked two developers so whoever is available can take it.

_P.S: I recommend turning on ` Hide whitespace changes` in the `Diff settings` while reviewing this PR as it'll make it much easier to see the actual layout changes._
